File: rgctl-tests/ecommerce-typescript/src/coolstore/models/catalogProduct.ts
/** Lightweight CoolStore catalog product (itemId keyed). */
export interface CatalogProduct {
  itemId: string;
  name: string;
  desc: string;
  price: number;
}

export function createCatalogProduct(
  itemId: string,
  name: string,
  desc: string,
  price: number,
): CatalogProduct {
  return { itemId, name, desc, price };
}

File: rgctl-tests/ecommerce-rust/src/coolstore/model/catalog_product.rs
use serde::{Deserialize, Serialize};

/// Lightweight CoolStore catalog product (itemId keyed).
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CatalogProduct {
    pub item_id: String,
    pub name: String,
    pub desc: String,
    pub price: f64,
}

impl CatalogProduct {
    pub fn new(item_id: impl Into<String>, name: impl Into<String>, desc: impl Into<String>, price: f64) -> Self {
        Self {
            item_id: item_id.into(),
            name: name.into(),
            desc: desc.into(),
            price,
        }
    }
}
